Src/Plugins/Security/SEC101_046.DiscordApiCredentialsValidator.cs

```diff
@@ -24,6 +22,11 @@ static DiscordApiCredentialsValidator()
             Instance = new DiscordApiCredentialsValidator();
         }
 
+        public static void Clear()
```

Comment:
>Clear [](http://example.com/codeflow?start=27&length=5)

Hello! This change is not what we discussed. You have added a new method. What I suggested what that you obtain a reference to the Instance property and null that out. Also, I suggested that we delete the static ctor and initialize this data in the IsValid helpers.

Reply:
yeah, i remember what we discussed...but, below my reasoning to not apply:

```
I was implementing the idea of static instance to be settable to null using reflection.

The issue: imagine a multi-threaded application, each thread can be in a different validation moment (let's just imagine that we have only one validator).

In one of the threads i decide to set to null.

This can generate multiple null references, because we are using the instance.dictionary or instance.hashseet or instance.validator
```